WolvenKit.Modkit/RED4/Tools/MorphTargetTools.cs

```diff
@@ -199,35 +199,44 @@ private static TargetsInfo GetTargetInfos(CR2WFile cr2w, int subMeshC)
             return targetsInfo;
         }
 
-        private static RawTargetContainer[] ContainRawTargets(MemoryStream diffsbuffer, MemoryStream mappingbuffer, uint[] numVertexDiffsInEachChunk, uint[] numVertexDiffsMappingInEachChunk, uint targetStartsInVertexDiffs, uint targetStartsInVertexDiffsMapping, Vec4 targetPositionDiffOffset, Vec4 targetPositionDiffScale, int subMeshC)
+        private static RawTargetContainer[] ContainRawTarget(MemoryStream diffsbuffer, MemoryStream mappingbuffer, uint[] numVertexDiffsInEachChunk, uint[] numVertexDiffsMappingInEachChunk, uint targetStartsInVertexDiffs, uint targetStartsInVertexDiffsMapping, Vec4 targetPositionDiffOffset, Vec4 targetPositionDiffScale, int subMeshC)
         {
+            var mappingByteSize = 2;
+            var mappingStartOffsetSizeMultiplier = mappingByteSize * 2; // num stored as half for some reason
+
             var rawtarget = new RawTargetContainer[subMeshC];
 
             var diffsbr = new BinaryReader(diffsbuffer);
             var mappingbr = new BinaryReader(mappingbuffer);
 
-            for (var i = 0; i < subMeshC; i++)
+            for (var subMeshIndex = 0; subMeshIndex < subMeshC; subMeshIndex++)
             {
-                var diffsCount = numVertexDiffsInEachChunk[i];
-                var vertexDelta = new TargetVec3[diffsCount];
-                var normalDelta = new Vec3[diffsCount];
-                var tangentDelta = new Vec3[diffsCount];
+                var diffsCount = numVertexDiffsInEachChunk[subMeshIndex];
+                var storedMapsCount = numVertexDiffsMappingInEachChunk[subMeshIndex];
 
-                if (i == 0)
-                {
-                    diffsbuffer.Position = targetStartsInVertexDiffs * 12;
-                }
-                else
-                {
-                    diffsbuffer.Position = targetStartsInVertexDiffs * 12;
+                // I *think* this will only ever be >, but just in case
```

Comment:
Not needed but `diffsCount / 2 - storedMapsCount`. Maybe could add a check then if its > 1 or < -1?